src/crypto/algorithms/megolm.js

```diff
@@ -706,24 +706,26 @@ MegolmDecryption.prototype.onRoomKeyEvent = function(event) {
         content.room_id, senderKey, forwardingKeyChain, sessionId,
         content.session_key, keysClaimed,
         exportFormat,
-    );
+    ).then(() => {
+        // cancel any outstanding room key requests for this session
+        this._crypto.cancelRoomKeyRequest({
+            algorithm: content.algorithm,
+            room_id: content.room_id,
+            session_id: content.session_id,
+            sender_key: senderKey,
+        });
 
-    // cancel any outstanding room key requests for this session
-    this._crypto.cancelRoomKeyRequest({
-        algorithm: content.algorithm,
-        room_id: content.room_id,
-        session_id: content.session_id,
-        sender_key: senderKey,
+        // have another go at decrypting events sent with this session.
+        this._retryDecryption(senderKey, sessionId);
+    }).catch((e) => {
+        console.error(`Error handling m.room_key_event: ${e}`);
     });
-
-    // have another go at decrypting events sent with this session.
-    this._retryDecryption(senderKey, sessionId);
 };
 
 /**
  * @inheritdoc
  */
-MegolmDecryption.prototype.hasKeysForKeyRequest = async function(keyRequest) {
+MegolmDecryption.prototype.hasKeysForKeyRequest = function(keyRequest) {
```

Comment:
They all need to be asynchronous. (The actual logic was that I wanted anything that touched sessionStore to be async, because I wanted to back the sessionStore with indexeddb, so sessionStore had to get an async interface).

Of course they can be asynchronous without being `async`. In this case: `hasInboundSessionKeys` is being changed to return a Promise, so the `async` here is redundant. I agree it could have stayed.